pixie/vm/stdlib.py

```diff
@@ -337,16 +337,36 @@ def apply__args(args):
 @returns(bool)
 @as_var("instance?")
 def _instance(c, o):
-    affirm(isinstance(c, Type), u"c must be a type")
+    from pixie.vm.persistent_vector import PersistentVector
+    if isinstance(c, Type):
+        return true if istypeinstance(o, c) else false
+    elif isinstance(c, PersistentVector):
```

Comment:
I'd prefer this be a check that something satisfies both ICounted and IIndexed. That way it supports things like arrays as well.